### PR TITLE
[MI100][TESTS] Fix issue 995 and 954: add some fp16 only test cases for asm igemm wrw/fwd nchw and remove W/A

### PR DESCRIPTION
**Pull request by shaojiewang**
Add regression test for #995 and #954 . 4 cases only applicable for fp16 or fp32. Remove W/A

### CODE REVIEW

**Review by atamazov — CHANGES_REQUESTED** [review]
This should be also resolved:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/097769a499d3971a6a9bf4f3cc37314f3fd8f149/test/CMakeLists.txt#L1156-L1157

**Review by atamazov — DISMISSED** [review]
Let me repeat:

This should be also resolved:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/097769a499d3971a6a9bf4f3cc37314f3fd8f149/test/CMakeLists.txt#L1156-L1157

I mean that the comment should be removed and the test case should be uncommented.

**Comment by shaojiewang** [comment]
> Let me repeat:
> 
> This should be also resolved:
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/097769a499d3971a6a9bf4f3cc37314f3fd8f149/test/CMakeLists.txt#L1156-L1157
> 
> I mean that the comment should be removed and the test case should be uncommented.

ok, understand.

**Comment by shaojiewang** [comment]
> Let me repeat:
> 
> This should be also resolved:
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/097769a499d3971a6a9bf4f3cc37314f3fd8f149/test/CMakeLists.txt#L1156-L1157
> 
> I mean that the comment should be removed and the test case should be uncommented.

This case is not applicable for --half mode. Move it to a float only test.

**Comment by shaojiewang** [comment]
Also remove W/A for 954.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1165,25 +1163,30 @@ COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIO
 COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  8  128 56 56 --weights 128 16 3 3 --pads_strides_dilations 1 1 1 1 1 1 --group-count 8 --disable-forward --disable-backward-weights
 )
 
+add_custom_test(test_conv_igemm_dynamic_xdlops_bwd_float SKIP_UNLESS_ALL HALF_DISABLED FLOAT_ENABLED GFX908_ENABLED VEGA_DISABLED
```
[the remark this answers is not in the document]
Comment (atamazov) [reply test/CMakeLists.txt:1166]: This code moved to gtest in #2409, bug still persists, details at https://github.com/ROCm/MIOpen/pull/2409/files#r1553637436

**Pull request merged**